cli/worker.rs

```diff
@@ -153,10 +165,33 @@ impl CliMainWorker {
     self.worker.dispatch_load_event(located_script_name!())?;
 
     loop {
-      self
-        .worker
-        .run_event_loop(maybe_coverage_collector.is_none())
-        .await?;
+      if let Some(hot_reload_manager) = maybe_hot_reload_manager.as_mut() {
+        let hmr_future =
+          hot_reload::run_hot_reload(hot_reload_manager).boxed_local();
+        let event_loop_future = self.worker.run_event_loop(false).boxed_local();
+
+        let result;
+        select! {
+          hmr_result = hmr_future => {
+            result = hmr_result;
+          },
+          event_loop_result = event_loop_future => {
+            result = event_loop_result;
+          }
+        }
+        if let Err(e) = result {
+          hot_reload_manager
+            .watcher_communicator
+            .change_restart_mode(WatcherRestartMode::Automatic);
```

Comment:
Line to refactor somehow to not access a dependency of a dependency.

Example:

```
 let communicator =
      self.shared.maybe_file_watcher_communicator.as_ref().unwrap();
 communicator.change_restart_mode(WatcherRestartMode::Automatic);
```

Or perhaps `maybe_hot_reload_manager` could contain the communicator as well.

Reply:
But `maybe_hot_reload_manager` already contains is as `hot_reload_manager.watcher_communicator`. I can change it to use `self.shared.maybe_file_watcher_communicator` though.